iac-common/src/test/java/org/sonar/iac/common/extension/IacSensorTest.java

```diff
@@ -71,6 +77,29 @@ void test_descriptor() {
     assertThat(sensorDescriptor.name()).isEqualTo("IaC Common Sensor");
   }
 
+
+  @Test
+  void test_descriptor_sonarqube_9_3() {
+    final boolean[] called = {false};
+    DefaultSensorDescriptor descriptor = new DefaultSensorDescriptor() {
+      public SensorDescriptor processesFilesIndependently() {
+        called[0] = true;
+        return this;
+      }
+    };
+    sensor(SonarRuntimeImpl.forSonarQube(Version.create(9, 3), SonarQubeSide.SCANNER, SonarEdition.DEVELOPER), null)
+      .describe(descriptor);
+    assertThat(called[0]).isTrue();
+  }
+
+  @Test
+  void test_descriptor_sonarqube_9_3_reflection_failure() {
+    DefaultSensorDescriptor descriptor = new DefaultSensorDescriptor();
+    sensor(SonarRuntimeImpl.forSonarQube(Version.create(9, 3), SonarQubeSide.SCANNER, SonarEdition.DEVELOPER), null)
+      .describe(descriptor);
+    assertThat(logTester.logs()).contains("Could not call SensorDescriptor.processesFilesIndependently() method");
```

Comment:
I don't really get the trick. Why don't we depend on latest sonar-plugin.jar?

Reply:
I don't quite see what we prove with those tests: just that, if there's a method with a given name, we will call it?
What about adding integration tests to ensure that we have a different behavior between SQ 9.2 and SQ dev?

Reply:
> Why don't we depend on latest sonar-plugin.jar?
I think it's because the sonar-plugin-9.3.jar is not yet released and we don't want to wait it to be released to update all our analyzers and finally get this feature in SQ 9.4

Reply:
Yeah. We have to do the reflection trick in the sensor as well as create our own implementation of DefaultSensorDescriptor in the tests because it's not available yet in the API.